cmd/template/nodepool/provider/aws.go

```diff
@@ -64,3 +149,59 @@ func WriteAWSTemplate(out io.Writer, config NodePoolCRsConfig) error {
 
 	return nil
 }
+
+func getCAPANodepoolTemplate(config NodePoolCRsConfig) (client.Template, error) {
```

Comment:
This function uses the upstream client to fetch the CAPA templates directly from the repo the way `clusterctl` does. The idea would be to extend this also to the other providers and versions than just `aws v20.0.0`

Reply:
What is the reason using this approach vs. getting the templates from apiextensions? Is this more convenient?

Reply:
Our idea was to stay close to the upstream and always get the right templates for the controller version and flavor we want to use. For now it seems convenient since we are still learning. Maybe in the future we want to have our own default templates.